src/components/Cluster/ClusterDetail/ClusterDetailView.js

```diff
@@ -142,19 +142,6 @@ class ClusterDetailView extends React.Component {
     return 'Not found';
   }
 
-  // Determine whether the current cluster can be upgraded
-  canClusterUpgrade() {
```

Comment:
Extracted this to clusterUtils so that I can unit test it.

Reply:
👍I am ok with this, but I think from now on it would be interesting to compute derived data on selectors, though.

Reply:
Ah yeah, I can make a selector! Only thing is I would probably still like an extracted function that the selector then uses, because the inputs of that function are a bit clearer (currentVersion, targetVersion, provider) instead of (state, props)

Reply:
Actually are there any rules about the inputs to selectors? We dont have to pass the top level state or props always right?

Reply:
We don't have rules for it. Some people say that it's best to pass top level state because if you normalize the store, then you just have to change arguments in one place. And I think that's what I am doing. 

In case of props I think it's best to pass the value you need instead of top level props, because props can  vary from one component to another.

Reply:
Ok I start to understand things more :)

Reply:
If you can take a look and let me know if the selector I made kinda makes sense, that would be great!

Reply:
I think it is ok! :-)